File: Regression/Simple Linear Regression/Simple Linear Regression.R

dataset = read.csv("Salary_Data.csv")

# install.packages("caTools")
library(caTools)
set.seed(123)

# split the dataset
split = sample.split(dataset$Salary, SplitRatio = 2/3)
trainingset = subset(dataset, split == TRUE)
testset = subset(dataset, split == F)

# Fitting the 
regressor = lm(formula = Salary ~ YearsExperience, data = trainingset)

# Predicing test set results
y_pred = predict(regressor, newdata = testset)

# Visualising results
# install.packages("ggplot2")
library(ggplot2)
ggplot() +
  geom_point(aes(x = trainingset$YearsExperience, y = trainingset$Salary),
             color = "red") +
  geom_line(aes(x = trainingset$YearsExperience, y = predict(regressor, newdata = trainingset)),
            color = "blue") +
  ggtitle("Salary VS Experience (Training set)") +
  xlab("Years of Experience") +
  ylab("Salary")

ggplot() +
  geom_point(aes(x = testset$YearsExperience, y = testset$Salary),
             color = "red") +
  geom_line(aes(x = trainingset$YearsExperience, y = predict(regressor, newdata = trainingset)),
            color = "blue") +
  ggtitle("Salary VS Experience (Test set)") +
  xlab("Years of Experience") +
  ylab("Salary")

File: Regression/Simple Linear Regression/Simple Linear Regression.py
from builtins import print

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

# Import the data set
dataset = pd.read_csv("Salary_Data.csv")
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Split the data set
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=1 / 3, random_state=0)
# Don't need feature scaling

# Fitting simple linear regression to the training set
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predicting the test set results
y_pred = regressor.predict(X_test)

# Visualise the training set results
plt.scatter(X_train, y_train, color="red")
plt.plot(X_train, regressor.predict(X_train), color="blue")
plt.title("Salary vs Experience (Training set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.show()

# Visualise the test set results
plt.scatter(X_test, y_test, color="red")
plt.plot(X_test, regressor.predict(X_test), color="blue")
plt.title("Salary vs Experience (Test set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.show()
